cocotb/binary.py

```diff
@@ -300,7 +314,9 @@ def _adjust_twos_comp(self, x):
         if self._n_bits is None:
             return x
         l = len(x)
-        if l <= self._n_bits:
+        if l == 0:
+            return x
+        elif l <= self._n_bits:
```

Comment:
```suggestion
        elif l < self._n_bits:
```
I think you will also need to apply the same change to `_adjust_signed_magnitude` and `_adjust_unsigned` as well. But maybe not? That code is a little hard to follow.

Reply:
I see the point of this change (make the else: statement work), but it has no connection to my change. Right?

Reply:
This change is not needed in `_adjust_unsigned` as there is no else: statement.

Reply:
I'll trust you on not needing the changes in the other ones. The change I suggested here was just because there is a `==` statement above, so there isn't a need to check for less then **or equals** here. It's not the end of the world though.